From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Sat, 1 Feb 2025 18:55:59 +0300
Subject: [PATCH] MSPT Tracking for each world


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index ddd0cab8f83a28b8f32d0a0f9f98cca74d53f23e..020165c7b0a8b921d2bbb3e26b83092d4748a981 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1850,7 +1850,16 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
     // DivineMC start - Parallel world ticking
     private void tickLevel(ServerLevel serverLevel, BooleanSupplier haveTime) {
         try {
+            // DivineMC start - MSPT Tracking for each world
+            long tickStart = Util.getNanos();
             serverLevel.tick(haveTime);
+
+            long tickDuration = Util.getNanos() - tickStart;
+
+            serverLevel.tickTimes5s.add(this.tickCount, tickDuration);
+            serverLevel.tickTimes10s.add(this.tickCount, tickDuration);
+            serverLevel.tickTimes60s.add(this.tickCount, tickDuration);
+            // DivineMC end - MSPT Tracking for each world
         } catch (Throwable levelTickingException) {
             CrashReport crashReport = CrashReport.forThrowable(levelTickingException, "Exception ticking world");
             serverLevel.fillReportDetails(crashReport);
diff --git a/net/minecraft/server/level/ServerLevel.java b/net/minecraft/server/level/ServerLevel.java
index de5787a8302deae67107a2e20e25a05083418eb1..b41ac21683a97dd9a43060e41e5344f5ac2deb1f 100644
--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -610,6 +610,12 @@ public class ServerLevel extends Level implements WorldGenLevel, ServerEntityGet
     }
     // Paper end - chunk tick iteration
 
+    // DivineMC start - MSPT Tracking for each world
+    public final MinecraftServer.TickTimes tickTimes5s = new MinecraftServer.TickTimes(100);
+    public final MinecraftServer.TickTimes tickTimes10s = new MinecraftServer.TickTimes(200);
+    public final MinecraftServer.TickTimes tickTimes60s = new MinecraftServer.TickTimes(1200);
+    // DivineMC end - MSPT Tracking for each world
+
     public ServerLevel(
         final MinecraftServer server,
         final Executor executor,
