--- a/net/minecraft/world/level/block/entity/HopperBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/HopperBlockEntity.java
@@ -425,6 +_,11 @@
 
         Direction direction = self.facing.getOpposite();
         if (isFullContainer(container, direction)) {
+            // DivineMC start - SparklyPaper: Allow throttling hopper checks if the target container is full
+            if (org.bxteam.divinemc.config.DivineConfig.PerformanceCategory.hopperThrottleWhenFull && org.bxteam.divinemc.config.DivineConfig.PerformanceCategory.hopperThrottleSkipTicks > 0) {
+                self.setCooldown(org.bxteam.divinemc.config.DivineConfig.PerformanceCategory.hopperThrottleSkipTicks);
+            }
+            // DivineMC end - SparklyPaper: Allow throttling hopper checks if the target container is full
             return false;
         }
 
