From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: NONPLAYT <76615486+NONPLAYT@users.noreply.github.com>
Date: Sat, 11 Jan 2025 23:22:12 +0300
Subject: [PATCH] Optimize default values for configs


diff --git a/src/main/java/org/purpurmc/purpur/PurpurConfig.java b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
index 241eb9a2aedbd6c5c3bf50716040545e586e81cc..eb553945b1f63116880ba1713c76fbec88626d06 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurConfig.java
@@ -246,7 +246,7 @@ public class PurpurConfig {
         laggingThreshold = getDouble("settings.lagging-threshold", laggingThreshold);
     }
 
-    public static boolean useAlternateKeepAlive = false;
+    public static boolean useAlternateKeepAlive = true; // DivineMC - Optimize default values for configs
     private static void useAlternateKeepAlive() {
         useAlternateKeepAlive = getBoolean("settings.use-alternate-keepalive", useAlternateKeepAlive);
     }
diff --git a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
index 6ee04e7cef3489096b2fd3c5ee81830bc01fca71..1322b750b834e5a1ee337d79766b2da6fc578716 100644
--- a/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
+++ b/src/main/java/org/purpurmc/purpur/PurpurWorldConfig.java
@@ -3419,8 +3419,8 @@ public class PurpurWorldConfig {
     public boolean villagerDisplayTradeItem = true;
     public int villagerSpawnIronGolemRadius = 0;
     public int villagerSpawnIronGolemLimit = 0;
-    public int villagerAcquirePoiSearchRadius = AcquirePoi.SCAN_RANGE;
-    public int villagerNearestBedSensorSearchRadius = AcquirePoi.SCAN_RANGE;
+    public int villagerAcquirePoiSearchRadius = 16; // DivineMC - Optimize default values for configs
+    public int villagerNearestBedSensorSearchRadius = 16; // DivineMC - Optimize default values for configs
     public Boolean villagerCanPickUpLoot = null;
     private void villagerSettings() {
         villagerRidable = getBoolean("mobs.villager.ridable", villagerRidable);
