--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -390,6 +_,7 @@
     public long activatedTick = Integer.MIN_VALUE;
     public boolean isTemporarilyActive;
     public long activatedImmunityTick = Integer.MIN_VALUE;
+    private boolean boundingBoxChanged = false; // DivineMC - VMP: skip entity move if movement is zero
 
     public void inactiveTick() {
     }
@@ -1167,6 +_,7 @@
     // Paper end - detailed watchdog information
 
     public void move(final MoverType moverType, Vec3 delta) {
+        if (!this.boundingBoxChanged && delta.equals(Vec3.ZERO)) return; // DivineMC - VMP: skip entity move if movement is zero
         final Vec3 originalMovement = delta; // Paper - Expose pre-collision velocity
         // Paper start - detailed watchdog information
         ca.spottedleaf.moonrise.common.util.TickThread.ensureTickThread("Cannot move an entity off-main");
@@ -4638,6 +_,7 @@
     }
 
     public final void setBoundingBox(final AABB bb) {
+        if (!this.bb.equals(bb)) this.boundingBoxChanged = true; // DivineMC - VMP: skip entity move if movement is zero
         // CraftBukkit start - block invalid bounding boxes
         double minX = bb.minX,
                 minY = bb.minY,
