--- a/purpur-api/build.gradle.kts
+++ b/purpur-api/build.gradle.kts
@@ -53,6 +53,7 @@ dependencies {
     api("org.apache.logging.log4j:log4j-api:$log4jVersion")
     api("org.slf4j:slf4j-api:$slf4jVersion")
     api("com.mojang:brigadier:1.3.10")
+    api("io.sentry:sentry:8.25.0") // DivineMC - Pufferfish: Sentry
 
     // Deprecate bungeecord-chat in favor of adventure
     api("net.md-5:bungeecord-chat:$bungeeCordChatVersion") {
@@ -92,6 +93,8 @@ dependencies {
     testImplementation("org.ow2.asm:asm-tree:9.9.1")
     mockitoAgent("org.mockito:mockito-core:5.22.0") { isTransitive = false } // configure mockito agent that is needed in newer java versions
     testRuntimeOnly("org.junit.platform:junit-platform-launcher:6.0.3")
+
+    compileOnly("org.apache.commons:commons-lang3:3.19.0") // DivineMC - Required for Bukkit
 }
 
 val generatedDir: java.nio.file.Path = rootProject.layout.projectDirectory.dir("paper-api/src/generated/java").asFile.toPath()
@@ -105,17 +108,21 @@ sourceSets {
         java {
             srcDir(generatedDir)
             srcDir(file("../paper-api/src/main/java"))
+            srcDir(file("../purpur-api/src/main/java")) // DivineMC - Rebrand
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
+            srcDir(file("../purpur-api/src/main/resources")) // DivineMC - Rebrand
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
+            srcDir(file("../purpur-api/src/test/java")) // DivineMC - Rebrand
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
+            srcDir(file("../purpur-api/src/test/resources")) // DivineMC - Rebrand
         }
     }
 }
@@ -237,7 +244,20 @@ tasks.withType<Javadoc>().configureEach {
             into("build/docs/javadoc")
         }
     }
+
+    options.addStringOption("Xdoclint:none", "-quiet") // DivineMC - Hide unnecessary javadoc warnings
+    options.addStringOption("-add-modules", "jdk.incubator.vector") // DivineMC - Required for simd
+}
+
+// DivineMC start - Hide unnecessary compilation warnings
+tasks.withType<JavaCompile> {
+    val compilerArgs = options.compilerArgs
+    compilerArgs.add("-Xlint:-module")
+    compilerArgs.add("-Xlint:-removal")
+    compilerArgs.add("-Xlint:-dep-ann")
+    compilerArgs.add("--add-modules=jdk.incubator.vector")
 }
+// DivineMC end - Hide unnecessary compilation warnings
 
 tasks.test {
     useJUnitPlatform()
